base/options.jl

```diff
@@ -34,11 +34,18 @@ struct JLOptions
     bindto::Ptr{UInt8}
     outputbc::Ptr{UInt8}
     outputunoptbc::Ptr{UInt8}
+    outputjitbc::Ptr{UInt8}
     outputo::Ptr{UInt8}
     outputji::Ptr{UInt8}
     incremental::Int8
 end
 
+# This runs early in the sysimage != is not defined yet
+if sizeof(JLOptions) === ccall(:jl_sizeof_jl_options, Int, ())
+else
+    ccall(:jl_throw, Void, (Any,), "Option structure mismatch")
```

Comment:
`throw` is a builtin function. you could ccall `jl_error` here to get the right Exception type, or just call throw